File: tape_equilibrium/tape_equilibrium.js
// https://app.codility.com/demo/results/trainingECH45R-US4/

function solution(A) {
  // write your code in JavaScript (Node.js 8.9.4)
  let totalSum = 0;

  for (let i = 0; i < A.length; i++) {
    totalSum = totalSum + A[i];
  }

  let minDiff = 1000 * 100000;
  let currentSum = 0;

  for (let i = 0; i < A.length - 1; i++) {
    currentSum = currentSum + A[i];
    let currentDiff = Math.abs(2 * currentSum - totalSum);

    if (currentDiff < minDiff) {
      minDiff = currentDiff;
    }
  }

  return minDiff;
}

File: tape_equilibrium/tape_equilibrium.py
# https://app.codility.com/demo/results/trainingBTHB6D-6BJ/

import sys


def solution(A):
    presum = []
    total_sum = 0

    for number in A:
        total_sum = total_sum + number
        presum.append(total_sum)

    minimal_diff = sys.maxsize

    for i in range(1, len(presum)):
        left = presum[i - 1]
        right = total_sum - left
        diff = abs(left - right)

        if (diff < minimal_diff):
            minimal_diff = diff

    return minimal_diff
